src/staticdata_utils.c

```diff
@@ -1118,14 +1124,23 @@ static void jl_insert_backedges(jl_array_t *edges, jl_array_t *ext_targets, jl_a
     // next build a map from external MethodInstances to their CodeInstance for insertion
     for (size_t i = 0; i < n_ext_cis; i++) {
         jl_code_instance_t *ci = (jl_code_instance_t*)jl_array_ptr_ref(ext_ci_list, i);
+        // assert(jl_egal(ci->owner, jl_nothing));
         assert(ci->min_world == minworld);
         if (ci->max_world == WORLD_AGE_REVALIDATION_SENTINEL) {
-            ptrhash_put(&cis_pending_validation, (void*)ci->def, (void*)ci);
+            void **bp = ptrhash_bp(&cis_pending_validation, (void*)ci->def);
+            if (*bp == HT_NOTFOUND)
+                *bp = (void*)ci;
+            else {
+                // XXX: How do we want to handle this?
+                ios_puts("DEV-WARN: Entering two CI for the same MI\n", ios_stderr);
+                jl_static_show(JL_STDERR, (jl_value_t*)ci->owner);
+                jl_static_show(JL_STDERR, (jl_value_t*)((jl_code_instance_t*) *bp)->owner);
+            }
```

Comment:
@vtjnash any ideas on this together with L249?

By adding the owner field I am changing the assumption the there is only one CI that is appropriate to serialize,
instead now there may be many that are still valid.

Is there a better way than me creating an array per MI?